ArduCopter/Copter.h

```diff
@@ -849,83 +904,75 @@ class Copter : public AP_HAL::HAL::Callbacks {
     void check_usb_mux(void);
     bool should_log(uint32_t mask);
     void set_default_frame_class();
-    void allocate_motors(void);
     uint8_t get_frame_mav_type();
     const char* get_frame_string();
+    void allocate_motors(void);
+
+    // takeoff.cpp
     bool current_mode_has_user_takeoff(bool must_navigate);
     bool do_user_takeoff(float takeoff_alt_cm, bool must_navigate);
     void takeoff_timer_start(float alt_cm);
     void takeoff_stop();
     void takeoff_get_climb_rates(float& pilot_climb_rate, float& takeoff_climb_rate);
-    void print_hit_enter();
+    void auto_takeoff_set_start_alt(void);
+    void auto_takeoff_attitude_run(float target_yaw_rate);
+
+    // terrain.cpp
+    void terrain_update();
+    void terrain_logging();
+    bool terrain_use();
+
+    // tuning.cpp
     void tuning();
-    void log_init(void);
-    void init_capabilities(void);
-    void dataflash_periodic(void);
-    void ins_periodic();
-    void accel_cal_update(void);
-    
-    uint16_t get_pilot_speed_dn();    
 
-#include "mode.h"
+    // UserCode.cpp
+    void userhook_init();
+    void userhook_FastLoop();
+    void userhook_50Hz();
+    void userhook_MediumLoop();
+    void userhook_SlowLoop();
+    void userhook_SuperSlowLoop();
 
-    Copter::Mode *flightmode;
+#include "mode.h"
```

Comment:
This is the only thing I don't like. Includes in the middle of a file is an ugly thing.

Reply:
Indeed, i wanted to move it up but I don't think we can.  This is an existing issue in master of course.